### PR TITLE
Add sample for AI extraction

### PR DESCRIPTION
**Pull request by ShammiL**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: reviewSummary/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+### Updating the Config.toml file
+```
+openAIKey = "<OPEN_AI_KEY>"
+```
+
+### Running the sample
+Run the following command from the root directory of the sample.
+```ballerina
+bal run
```
Comment (SasinduDilshara) [inline reviewSummary/README.md]:
```suggestion
$ bal run
```

File: reviewSummary/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+### Send a request
+```
+POST http://localhost:8080/api/reviews/summary
+
+{
+  "reviews": ["I recently ordered a laptop and it was in very good condition and it was delivered it on time.
+    But there were some software glitches and I couldn't find enough instructions to troubleshoot them.
+    Would be great if they provided more detailed manuals.",
+    "The smartphone I ordered arrived with a damaged screen. They need to be careful when packaging.",
+    "Extrmely satisfied with my orders. Would suggest this place to anyone.",
```
Comment (SasinduDilshara) [inline reviewSummary/README.md]:
```suggestion
    "Extremely satisfied with my orders. Would suggest this place to anyone.",
```

File: reviewSummary/main.bal [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+service /api/reviews on new http:Listener(8080) {
+    resource function post summary(SummaryRequest summaryRequest) returns error? {
+        chat:CreateChatCompletionRequest request = {
+            model: "gpt-3.5-turbo",
+            messages: [
+                {
+                    role: "user",
+                    content: string `
+                    Extract the following details in JSON from the reviews given.
```
Comment (SasinduDilshara) [inline reviewSummary/main.bal]: Shall we update the formatting

File: reviewSummary/main.bal [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                    The fields should contain points extracted from all reviews
+                    Here are the reviews:
+                    ${string:'join(",", ...summaryRequest.reviews)}
+                    `
+                }
+            ]
+        };
+        do {
+            chat:CreateChatCompletionResponse SummaryResponse = check openAiChat->/chat/completions.post(request);
+            if SummaryResponse.choices.length() < 1 {
```
Comment (SasinduDilshara) [inline reviewSummary/main.bal]:
```suggestion
            if SummaryResponse.choices.length() == 0 {
```

File: reviewSummary/tests/test-01.http [hunk 5]
```diff
@@ -0,0 +1,11 @@
+POST http://localhost:8080/api/reviews/summary
+
+{
+  "reviews": ["I recently ordered a laptop and it was in very good condition and it was delivered it on time.
+    But there were some software glitches and I couldn't find enough instructions to troubleshoot them.
+    Would be great if they provided more detailed manuals.",
+    "The smartphone I ordered arrived with a damaged screen. They need to be careful when packaging.",
+    "Extrmely satisfied with my orders. Would suggest this place to anyone.",
```
Comment (SasinduDilshara) [inline reviewSummary/tests/test-01.http]:
```suggestion
    "Extremely satisfied with my orders. Would suggest this place to anyone.",
```

File: reviewSummary/.devcontainer.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (pcnfernando) [inline reviewSummary/.devcontainer.json]: We can remove this

File: reviewSummary/Ballerina.toml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+[package]
+org = "etl"
+name = "reviewSummary"
+version = "0.1.0"
+distribution = "2201.8.0"
+
+[build-options]
```
Comment (pcnfernando) [inline reviewSummary/Ballerina.toml]: Can remove this

File: reviewSummary/main.bal [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+                        The fields should contain points extracted from all reviews
+                        Here are the reviews:
+                        ${string:'join(",", ...summaryRequest.reviews)}
+                    `
+                }
+            ]
+        };
+        do {
+            // Call the OpenAI chat completion endpoint and retrive chat completion response.
+            chat:CreateChatCompletionResponse SummaryResponse = check openAiChat->/chat/completions.post(request);
```
Comment (pcnfernando) [inline reviewSummary/main.bal]:
```suggestion
            chat:CreateChatCompletionResponse summary = check openAiChat->/chat/completions.post(request);
```

Comment (pcnfernando) [reply reviewSummary/main.bal]: Do the required changes as per this change

File: reviewSummary/README.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+### Running the sample
+Run the following command from the root directory of the sample.
+```ballerina
+$ bal run
+```
+
+### Send a request
+```
+POST http://localhost:8080/api/reviews/summary
```
Comment (pcnfernando) [inline reviewSummary/README.md]: Let's have the actual curl here. Check other exammples as well

File: reviewSummary/Ballerina.toml [hunk 10]
```diff
@@ -0,0 +1,5 @@
+[package]
+org = "etl"
+name = "reviewSummary"
```
Comment (pcnfernando) [inline reviewSummary/Ballerina.toml:3]: Use the same conversion as the others

File: reviewSummary/README.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
+## Steps to run the sample
+
+### 1. Configuring the Config.toml file
+1. Create a `Config.toml` file in the root directory and add the below content.
+```toml
+openAIKey = "<OPEN_AI_KEY>"
+```
+2. Create an [OpenAI account](https://platform.openai.com) and obtain 
+an [API key](https://platform.openai.com/account/api-keys) and replace with `<OPEN_AI_KEY>`.
```
Comment (pcnfernando) [inline reviewSummary/README.md:9]:
```suggestion
an [API key](https://platform.openai.com/account/api-keys) and use it as `<OPEN_AI_KEY>` in the `Config.toml`.
```

File: reviewSummary/main.bal [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+
+configurable string openAIKey = ?;
+
+type SummaryRequest record {|
+    string[] reviews;
+|};
+
+final chat:Client openAiChat = check new ({auth: {token: openAIKey}});
+
+service /api/reviews on new http:Listener(8080) {
```
Comment (pcnfernando) [inline reviewSummary/main.bal:13]: Mark these isolated

**Review by pcnfernando — APPROVED** [review]
Suggestions will be handled separately. Hence approving

**Pull request merged**